webapp/tests/data_access/test_form_data_controller.py

```diff
@@ -0,0 +1,34 @@
+import unittest
+from unittest.mock import patch
+
+import fakeredis as fakeredis
+
+from pydantic import MissingError
+from app.data_access.form_data_controller import save, get
+from app.config import Config
+
+redis_fake = fakeredis.FakeStrictRedis()
+
+
+@patch("app.data_access.form_data_controller.r", redis_fake)
+class TestFormDataController(unittest.TestCase):
+
+    key = "key"
+    value = "value"
+    ttl_seconds = Config.SESSION_DATA_REDIS_TTL_HOURS * 3600
+
+    def tearDown(self):
+        redis_fake.flushall()
+
+    def test_if_saving_key_value_then_return_true(self):
+        saved = save(self.key, self.value)
+        self.assertTrue(self.ttl_seconds == redis_fake.ttl(self.key))
+        self.assertTrue(saved)
```

Comment:
I would split this test into two tests. If possible a test should only test one thing:
- test_if_key_value_provided_then_save_key_value_pair_into_database
- test_if_key_value_provided_then_set_correct_configuration_ttl_to_database_entry

Reply:
You could also just test if the correct underlying function is called with the correct parameters to keep it a "true" unit test. Same for the other test